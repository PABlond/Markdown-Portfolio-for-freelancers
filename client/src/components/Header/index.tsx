import React from "react"
import { connect } from "react-redux"

const Header = ({ header }: any) => {
  return <div id="header" dangerouslySetInnerHTML={{__html: header}} />
}

const mapStateToProps = (state: any) => {
  const { header } = state.content
  return {
    header,
  }
}

export default connect(mapStateToProps)(Header)
