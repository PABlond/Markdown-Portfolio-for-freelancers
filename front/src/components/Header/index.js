import React from "react"
import store from "./../../state/store"
import pageStructure from "../App/pageStructure"

export default ({ content }) => {
  const { header } = store.getState().content
  return (
    <div id={pageStructure[content.n].id}>
      <div>
        <h1>{header.name}</h1>
        <h2>{header.title}</h2>
      </div>
      <p>{header.subtitle}</p>
    </div>
  )
}
