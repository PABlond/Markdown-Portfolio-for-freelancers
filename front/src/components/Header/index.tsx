import React from "react"
import store from "../../state/store"

export default () => {
  const {
    header,
  }: {
    header: { name: string; title: string; subtitle: string }
  } = store.getState().content
  return (
    <div id="header">
      <div>
        <h1>{header.name}</h1>
        <h2>{header.title}</h2>
      </div>
      <p>{header.subtitle}</p>
    </div>
  )
}
