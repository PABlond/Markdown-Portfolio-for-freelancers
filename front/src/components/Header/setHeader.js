import React, { useState, useEffect } from "react"
import store from "./../../state/store"
import { Container, Row, Button, Form } from "react-bootstrap"
import setHeader from "./../../services/setHeader"
import dispatchFullContent from "./../../state/actions/dispatchFullContent"

export default () => {
  const props = store.getState()
  const { header } = props.content
  const [headerState, setHeaderState] = useState(header)

  useEffect(() => {
    const { about, works, certifications } = props.content
    dispatchFullContent({ works, certifications, header: headerState, about })
  }, [headerState])

  const updateHeader = e =>
    setHeaderState({
      ...headerState,
      [e.target.name]: e.target.value,
    })

  const onSubmit = async e => {
    e.preventDefault()
    const { name, title, subtitle } = headerState
    await setHeader({ name, title, subtitle })
  }

  return (
    <Container>
      <Form onSubmit={onSubmit}>
        <Form.Group controlId="formBasicEmail">
          <Form.Label>Name</Form.Label>
          <Form.Control
            type="text"
            name="name"
            value={headerState.name}
            onChange={updateHeader}
          />
        </Form.Group>
        <Form.Group controlId="formBasicEmail">
          <Form.Label>Title</Form.Label>
          <Form.Control
            type="text"
            name="title"
            value={headerState.title}
            onChange={updateHeader}
          />
        </Form.Group>
        <Form.Group controlId="formBasicEmail">
          <Form.Label>Subtitle</Form.Label>
          <Form.Control
            type="text"
            name="subtitle"
            value={headerState.subtitle}
            onChange={updateHeader}
          />
        </Form.Group>
        <Button variant="primary" type="submit">
          Submit
        </Button>
      </Form>
    </Container>
  )
}
