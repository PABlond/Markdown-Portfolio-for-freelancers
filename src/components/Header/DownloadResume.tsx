import React from "react"
import { StaticQuery, graphql } from "gatsby"
import formatResumePage from "./../../helpers/resumeHelpers"
import Resume from "./../Resume"
import { PDFDownloadLink } from "@react-pdf/renderer"
import { connect } from "react-redux"
import { dispatchResume } from "./../../config/constants"
import FileSaver from "file-saver"
import { IState } from "./../../interfaces/state.interface"

const DownloadResume = ({
  dispatchFullResume,
}: {
  dispatchFullResume: (payload: IState["resume"]) => void
}) => {
  return (
    <>
      <StaticQuery
        query={graphql`
          query GetResumeData {
            allMarkdownRemark {
              edges {
                node {
                  html
                  frontmatter {
                    title
                  }
                }
              }
            }
          }
        `}
        render={data => {
          const formatedData = formatResumePage(data.allMarkdownRemark.edges)
          dispatchFullResume(formatedData)

          return (
            <PDFDownloadLink document={<Resume />} fileName="somename.pdf">
              {({ blob, url, loading, error }) => {
                if (error) console.log(error)
                if (url) console.log(url)
                return (
                  !loading && FileSaver.saveAs(blob as any, "Pierre-Alexis_Blond_Resume.pdf")
                )
              }}
            </PDFDownloadLink>
          )
        }}
      />
    </>
  )
}

const mapDispatchToProps = (dispatch: any) => ({
  dispatchFullResume: (payload: IState["resume"]) =>
    dispatch({ type: dispatchResume, payload }),
})

export default connect(
  null,
  mapDispatchToProps
)(DownloadResume)
