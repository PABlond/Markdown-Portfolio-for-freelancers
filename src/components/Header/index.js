import React from "react"
import { Random } from "react-animated-text"
import pageStructure from "../App/pageStructure"

export default ({ content }) => {
  return (
    <div id={pageStructure[content.n].id}>
      <div>
        <h1>
          <Random
            text={"Pierre-Alexis Blond"}
            iterations={1}
            effect="verticalFadeIn"
            effectChange={2.0}
            effectDirection="up"
          />
        </h1>
        <h2>
          <Random
            text="Full Stack developer"
            effect="verticalFadeIn"
            effectChange={2.2}
            delay={5}
            speed={10}
            iterations={1}
          />
        </h2>
      </div>
      <p dangerouslySetInnerHTML={content} />
    </div>
  )
}
