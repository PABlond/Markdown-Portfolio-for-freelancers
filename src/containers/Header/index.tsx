import React from "react"
import { header as headerContent } from "../../data/content"
import { StaticQuery, graphql } from "gatsby"
import Img, { FluidObject } from "gatsby-image"

export default () => {
  const { name, role, description, resume } = headerContent
  return (
    <section id="header">
      <div id="header-heading" className="animated fadeInDownBig">
        <h1>{name}</h1>
        <h2>{role}</h2>
      </div>
      <div id="header-img">
        <StaticQuery
          query={graphql`
            query HeaderImg {
              file(relativePath: { eq: "header.jpeg" }) {
                id
                childImageSharp {
                  fluid(maxWidth: 700) {
                    ...GatsbyImageSharpFluid_noBase64
                  }
                }
              }
            }
          `}
          render={(data: {
            file: { childImageSharp: { fluid: FluidObject } }
          }) => <Img {...data.file.childImageSharp} />}
        />
      </div>

      <div className="row">
        <div id="header-text" className="offset-md-2 col-md-8">
          <p>{description}</p>
          <a href="/resume_Pierre-Alexis_Blond.pdf" download>{resume}</a>
        </div>
      </div>
    </section>
  )
}
