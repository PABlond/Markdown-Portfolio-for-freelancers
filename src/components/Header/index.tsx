import React, { useState } from "react"
import { connect } from "react-redux"
import { IHeader } from "./../../interfaces/module.interface"
import DownloadResume from "./DownloadResume"
import parse, { domToReact } from "html-react-parser"

const Header = ({ header }: IHeader) => {
  const [isDL, setIsDl] = useState<Boolean>(false)
  const options = {
    replace: ({ attribs, children }: any) => {
      if (!attribs) return

      if (attribs.id === "header-text") {
        console.log(attribs, children)
        return (
          <div id="header-text" className="offset-md-2 col-md-8">
            {domToReact(children)}
            <a
              onClick={() => {
                setTimeout(() => setIsDl(false), 1000)
                return setIsDl(true)
              }}
            >{`-> My Resume <-`}</a>
          </div>
        )
      }
    },
  }
  return (
    <>
      <div id="header">{parse(header.__html as string, options)}</div>
      {isDL && <DownloadResume />}
    </>
  )
}

const mapStateToProps = (state: any) => {
  const { header } = state.content
  return {
    header,
  }
}

export default connect(mapStateToProps)(Header)
