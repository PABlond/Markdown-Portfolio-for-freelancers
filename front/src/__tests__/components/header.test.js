import React from "react"
import renderer from "react-test-renderer"
import { shallow, configure } from "enzyme"
import Header from "../../components/Header"
import Adapter from "enzyme-adapter-react-16"

configure({ adapter: new Adapter() })

const constactShallowCfg = () => {
  const props = {
    content: { n: "1" },
  }
  const enzymeWrapper = shallow(<Header {...props} />)
  return {
    props,
    enzymeWrapper,
  }
}

describe("Shallow rendered Contact", () => {
  it("should render match snapshot", () => {
    const { props } = constactShallowCfg()
    const component = renderer.create(<Header {...props} />)
    let tree = component.toJSON()
    console.log(tree)
    expect(tree).toMatchSnapshot()
  })
  it("should render contact as expected", () => {
    const { enzymeWrapper } = constactShallowCfg()
    expect(enzymeWrapper.find("h1").text()).toBe("")
    expect(enzymeWrapper.find('div').first().prop('id')).toBe('header')
  })
})
