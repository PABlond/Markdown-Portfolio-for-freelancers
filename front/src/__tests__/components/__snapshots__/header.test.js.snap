// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow rendered Contact should render match snapshot 1`] = `
<div
  id="header"
>
  <div>
    <h1>
      
    </h1>
    <h2>
      
    </h2>
  </div>
  <p>
    
  </p>
</div>
`;
